Extension/src/LanguageServer/configurations.ts

```diff
@@ -71,7 +71,7 @@ export interface Configuration {
     cStandardIsExplicit?: boolean;
     cppStandard?: string;
     cppStandardIsExplicit?: boolean;
-    includePath?: string[];
+    includePath?: string[] | null;
```

Comment:
The includePath/forcedInclude should not have the `| null` -- you can get around that by changing the return value of updateConfigurationPathArray to `string[] | undefined {` and adding 

```
        if (paths === null) {
            return undefined;
        }
```

Reply:
And I think compilerPathInCppPropertiesJson and compilerPath should be `| null` since those are the defaults in package.json.

Reply:
Changing the compilerPath to allow null just means you have to add `| null` in a few spots or use something like `=== null ? undefined : compilerPath` to convert the null to an undefined if being assigned to something that doesn't accept a null.

Reply:
Let's just try to get typings correct for the settings objects in this PR and then we can look at whether there are any bugs that need to be fixed. I'd like to come back to this issue in a subsequent PR.